File: app/src/main/java/me/imzack/app/ender/injector/component/TypeSearchComponent.java
package me.imzack.app.ender.injector.component;

import me.imzack.app.ender.injector.module.TypeSearchPresenterModule;
import me.imzack.app.ender.injector.scope.ActivityScope;
import me.imzack.app.ender.view.activity.TypeSearchActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = TypeSearchPresenterModule.class)
public interface TypeSearchComponent {

    void inject(TypeSearchActivity typeSearchActivity);
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/TypeSearchComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.TypeSearchPresenterModule
import net.zackzhang.app.end.injector.scope.ActivityScope
import net.zackzhang.app.end.view.activity.TypeSearchActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(TypeSearchPresenterModule::class))
interface TypeSearchComponent {

    fun inject(typeSearchActivity: TypeSearchActivity)
}
